lib/resources/config.js

```diff
@@ -67,11 +69,46 @@ module.exports = (service, endpoint) => {
         ? google.auth.getToken(body.code)
           .catch((error) => reject(Problem.user.oauth({ reason: `Unexpected error received: ${printPairs(error.response.data)}` })))
           .then((result) => Promise.all([
-            Configs.set('backups.main', JSON.stringify({ type: 'google', keys: actor.meta.keys })),
-            Configs.set('backups.google', JSON.stringify(result.tokens))
+            Configs.set('backups.main', { type: 'google', keys: actor.meta.keys }),
+            Configs.set('backups.google', result.tokens)
           ]))
           .then(() => Actors.consume(actor))
           .then(success)
         : Problem.user.oauth('No authorization code was given. Did you copy and paste the text from the Sign In page?')))));
+
+  service.post('/config/analytics', endpoint(({ Configs }, { auth, body }) =>
+    auth.canOrReject('config.set', Config.species)
+      .then(() => {
+        const value = { enabled: !!body.enabled };
+        if (value.enabled) {
+          if (isPresent(body.email)) value.email = body.email.toString();
+          if (isPresent(body.organization))
+            value.organization = body.organization.toString();
+        }
+        return Configs.set('analytics', value);
+      })));
+
+  const checkKey = (key) => () => {
+    if (reservedKeys.some((k) => key === k || key.startsWith(`${k}.`)))
+      throw Problem.user.unexpectedValue({
+        field: 'key',
+        value: key,
+        reason: 'the key is reserved for internal use.'
+      });
+  };
+
+  service.get('/config/:key', endpoint(({ Configs }, { auth, params }) =>
+    auth.canOrReject('config.read', Config.species)
+      .then(checkKey(params.key))
+      .then(() => Configs.get(params.key))
+      .then(getOrNotFound)));
+
+  service.delete('/config/:key', endpoint(({ Configs }, { auth, params }) =>
+    auth.canOrReject('config.set', Config.species)
```

Comment:
Right now, I'm using a single verb for both POST and DELETE: `config.set`. However, maybe I should use two verbs, `config.set` and `config.unset`? That would seem to nicely parallel the query methods `Configs.set()` and `Configs.unset()`. (Note that I haven't added any new verbs to the `admin` role yet.)

Reply:
i don't feel like there will ever be a meaningful distinction between set and unset at the site config level.
maybe if  we were talking about eg change vs clear some like project level flag that has a fallback from a higher level or something.

Reply:
The new commit continues to use a single verb: `config.set`. I've added logging for the existing query method `Configs.unset()`, logging a `config.set` action.